airflow/providers/zendesk/hooks/zendesk.py

```diff
@@ -31,90 +32,64 @@ class ZendeskHook(BaseHook):
+        conn = self.get_connection(self.zendesk_conn_id)
+        url = "https://" + conn.host
+        domain = conn.host
+        subdomain: Optional[str] = None
+        if conn.host.count(".") >= 2:
+            dot_splitted_string = conn.host.rsplit(".", 2)
+            subdomain = dot_splitted_string[0]
+            domain = ".".join(dot_splitted_string[1:])
+        return Zenpy(domain=domain, subdomain=subdomain, email=conn.login, password=conn.password), url
 
-    def __handle_rate_limit_exception(self, rate_limit_exception: ZendeskError) -> None:
+    def get_conn(self) -> Zenpy:
         """
-        Sleep for the time specified in the exception. If not specified, wait
-        for 60 seconds.
+        Get the underlying Zenpy client.
+
+        :return: zenpy.Zenpy client.
         """
-        retry_after = int(rate_limit_exception.response.headers.get('Retry-After', 60))
-        self.log.info("Hit Zendesk API rate limit. Pausing for %s seconds", retry_after)
-        time.sleep(retry_after)
+        return self.zenpy_client
 
     def call(
         self,
         path: str,
         query: Optional[dict] = None,
-        get_all_pages: bool = True,
-        side_loading: bool = False,
-    ) -> dict:
+        **kwargs: Any,
+    ) -> BaseObject:
         """
-        Call Zendesk API and return results
+        Call Zendesk API and return results.
 
-        :param path: The Zendesk API to call
-        :param query: Query parameters
-        :param get_all_pages: Accumulate results over all pages before
-               returning. Due to strict rate limiting, this can often timeout.
-               Waits for recommended period between tries after a timeout.
-        :param side_loading: Retrieve related records as part of a single
-               request. In order to enable side-loading, add an 'include'
-               query parameter containing a comma-separated list of resources
-               to load. For more information on side-loading see
-               https://developer.zendesk.com/rest_api/docs/core/side_loading
-        """
-        query_params = query or {}
-        zendesk = self.get_conn()
-        first_request_successful = False
-
-        while not first_request_successful:
-            try:
-                results = zendesk.call(path, query_params)
-                first_request_successful = True
-            except RateLimitError as rle:
-                self.__handle_rate_limit_exception(rle)
-
-        # Find the key with the results
-        keys = [path.split("/")[-1].split(".json")[0]]
-        next_page = results['next_page']
-        if side_loading:
-            keys += query_params['include'].split(',')
-        results = {key: results[key] for key in keys}
+        This endpoint is kept for backward compatibility purpose but it should be
+        removed in the future to expose specific methods for each resource.
+        TODO: When removing this method, remove the reference to _get in __init__
```

Comment:
I agree. As mentioned before this is a private function so we can remove it.
Can you add a single resource? It would be easier for others to add more methods when the first one available.

Reply:
Done